cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -100,6 +100,20 @@ private State(ProgramRunStatus runStatus) {
       this.runStatus = runStatus;
     }
 
+    /**
+     * Gives the {@link ProgramController.State} for a given {@link ProgramRunStatus}
+     * @param status : the status
+     * @return the state for the status or null if there is no defined state for the given status
+     */
+    public static State getControllerStateByStatus (ProgramRunStatus status) {
```

Comment:
better to put this in the ProgramController.State enum.  Also, this is a very weird method, there are many states that map to ProgramRunStatus.RUNNING.  This one will return STARTING all the time. Is that what you want?